### ISSUE TITLE
lambdify() with ** 2 gives incorrect result

### ISSUE THREAD

**Issue by sorrge** [sympy/sympy#27235]
The code:
```
import sympy as sp

x = sp.symbols("x")
ff = sp.sech(x) ** 2
print(ff.subs(x, 3).evalf())
print(sp.lambdify(x, ff)(3))
```
prints different values for evalf() and lambdify(): 0.00986603716544019 and 10.067661995777765. The evalf() result is correct. If you remove the "** 2" from the expression, the results match.

**Comment by bjodah** [comment]
What version of `sympy` are you using? I cannot reproduce your findings.

**Comment by oscarbenjamin** [comment]
I get the same result from both on master so I think this was already fixed:
```python
In [19]: import sympy as sp
    ...: 
    ...: x = sp.symbols("x")
    ...: ff = sp.sech(x) ** 2
    ...: print(ff.subs(x, 3).evalf())
    ...: print(sp.lambdify(x, ff)(3))
0.00986603716544019
0.00986603716544019
```
I can reproduce this with SymPy 1.12 though:
```python
In [1]: import sympy as sp
   ...: 
   ...: x = sp.symbols("x")
   ...: ff = sp.sech(x) ** 2
   ...: print(ff.subs(x, 3).evalf())
   ...: print(sp.lambdify(x, ff)(3))
0.00986603716544019
10.067661995777765
```
The fix was 83f8c53cbc7239e46c2c189bfbd4d28131f80b9e from gh-25514.

The generated code was changed from this:
```
def _lambdifygenerated(x):
    return ((1/2)*exp(x) + (1/2)*exp(-x))**(-1.0)**2
```
to this:
```
def _lambdifygenerated(x):
    return (((1/2)*exp(x) + (1/2)*exp(-x))**(-1.0))**2
```
The additional brackets make it correct.

A similar test for `sec` was add in gh-25514 so I think this can be closed.

Thanks for reporting.
